spring-cloud-gcp-secretmanager/src/main/java/org/springframework/cloud/gcp/secretmanager/SecretManagerTemplate.java

```diff
@@ -0,0 +1,147 @@
+		this.projectIdProvider = projectIdProvider;
+	}
+
+	@Override
+	public void createSecret(String secretId, String payload) {
+		if (!secretExists(secretId)) {
+			createSecret(secretId);
+		}
+
+		createNewSecretVersion(secretId, ByteString.copyFromUtf8(payload));
+	}
+
+	@Override
+	public void createSecret(String secretId, byte[] payload) {
+		if (!secretExists(secretId)) {
+			createSecret(secretId);
+		}
+
+		createNewSecretVersion(secretId, ByteString.copyFrom(payload));
+	}
+
+	@Override
+	public String getSecretString(String secretId, String versionName) {
+		return getSecretByteString(secretId, versionName).toStringUtf8();
+	}
+
+	@Override
+	public byte[] getSecretBytes(String secretId, String versionName) {
+		return getSecretByteString(secretId, versionName).toByteArray();
+	}
+
+	@Override
+	public ByteString getSecretByteString(String secretId, String versionName) {
+		SecretVersionName secretVersionName = SecretVersionName.of(
+				this.projectIdProvider.getProjectId(),
+				secretId,
+				versionName);
+
+		AccessSecretVersionResponse response = secretManagerServiceClient.accessSecretVersion(secretVersionName);
+
+		return response.getPayload().getData();
+	}
+
+	/**
+	 * Create a new version of the secret with the specified payload under a {@link Secret}.
+	 */
+	private void createNewSecretVersion(String secretId, ByteString byteStringPayload) {
+		SecretName name = SecretName.of(projectIdProvider.getProjectId(), secretId);
+		SecretPayload payloadObject = SecretPayload.newBuilder()
+				.setData(byteStringPayload)
+				.build();
+
+		AddSecretVersionRequest payloadRequest = AddSecretVersionRequest.newBuilder()
+				.setParent(name.toString())
+				.setPayload(payloadObject)
+				.build();
+		secretManagerServiceClient.addSecretVersion(payloadRequest);
+	}
+
+	/**
+	 * Creates a new secret for the GCP Project.
+	 *
+	 * <p>
+	 * Note that the {@link Secret} object does not contain the secret payload. You must
+	 * create versions of the secret which stores the payload of the secret.
+	 */
+	private void createSecret(String secretId) {
+		ProjectName projectName = ProjectName.of(projectIdProvider.getProjectId());
+
+		Secret secret = Secret.newBuilder()
+				.setReplication(
+						Replication.newBuilder()
+								.setAutomatic(Replication.Automatic.newBuilder().build())
+								.build())
+				.build();
+		CreateSecretRequest request = CreateSecretRequest.newBuilder()
+				.setParent(projectName.toString())
+				.setSecretId(secretId)
+				.setSecret(secret)
+				.build();
+		this.secretManagerServiceClient.createSecret(request);
+	}
+
+	/**
+	 * Returns true if there already exists a secret under the GCP project with the
+	 * {@code secretId}.
+	 */
+	private boolean secretExists(String secretId) {
+		ProjectName projectName = ProjectName.of(this.projectIdProvider.getProjectId());
+		ListSecretsPagedResponse listSecretsResponse = this.secretManagerServiceClient.listSecrets(projectName);
```

Comment:
This can be a lot of secrets.
Can [getSecret()](https://github.com/googleapis/java-secretmanager/blob/d4f471f8b3ad5d2d24f848a886a4a6dd40231500/google-cloud-secretmanager/src/main/java/com/google/cloud/secretmanager/v1beta1/SecretManagerServiceClient.java#L546) be used instead to retrieve the specific secret?

Reply:
Done! The getSecret approach is now being used; requires a try/catch style.